components/InlineDropDownMenu/InlineDropDownMenu.js

```diff
@@ -0,0 +1,130 @@
+import React, { PureComponent } from 'react';
+import autoBindReact from 'auto-bind/react';
+import { FlatList, Animated } from 'react-native';
+import _ from 'lodash';
+import PropTypes from 'prop-types';
+import { connectStyle } from '@shoutem/theme';
+import { TouchableOpacity } from '../TouchableOpacity';
+import { View } from '../View';
+import { Caption, Text } from '../Text';
+import { Icon } from '../Icon';
+import { InlineDropDownMenuItem } from './InlineDropDownMenuItem';
+
+const AnimatedIcon = Animated.createAnimatedComponent(Icon);
+
+const optionShape = PropTypes.shape({
+  title: PropTypes.string,
+  key: PropTypes.string,
+});
+
+class InlineDropDownMenu extends PureComponent {
+  static propTypes = {
+    heading: PropTypes.string,
+    selectedDescriptor: PropTypes.string,
+    options: PropTypes.arrayOf(optionShape).isRequired,
+    onOptionSelected: PropTypes.func,
+    selectedOption: optionShape,
+  };
+
+  constructor(props) {
+    super(props);
+
+    autoBindReact(this);
+
+    this.dropDownIconValue = new Animated.Value(0);
+
+    this.state = {
+      collapsed: false,
+    };
+  }
+
+  handleToggleMenuPress() {
+    const { collapsed } = this.state;
+
+    const toValue = collapsed ? 0 : 1;
+
+    this.setState(
+      { collapsed: !collapsed },
+      () => Animated.timing(
+        this.dropDownIconValue,
+        {
+          toValue,
+          useNativeDriver: true,
+          duration: 300,
+        }
+      ).start());
+  }
+
+  handleOptionPress(option) {
+    const { onOptionSelected } = this.props;
+
+    if (onOptionSelected) {
+      onOptionSelected(option);
+    }
+
+    this.handleToggleMenuPress();
+  }
+
+  renderOption({ item, index }) {
+    const { selectedOption, selectedDescriptor } = this.props;
+
+    const isSelected = selectedOption.key === item.key;
+
+    return (
+      <InlineDropDownMenuItem
+        item={item}
+        isSelected={isSelected}
+        index={index}
+        selectedDescriptor={selectedDescriptor}
+        onItemPressed={this.handleOptionPress}
+      />
+    )
+  }
+
+  render() {
+    const {
+      style,
+      heading,
+      selectedOption,
+      options,
+    } = this.props;
+    const { collapsed } = this.state;
+
+    return (
+      <View>
+        <TouchableOpacity style={style.container} onPress={this.handleToggleMenuPress}>
+          <Caption styleName="muted md-gutter-bottom">{heading}</Caption>
+          <View styleName="space-between horizontal v-center">
+            <Text>{selectedOption?.title}</Text>
```

Comment:
`selectedOption?.title` Smexy.

Although, won't this result in a Text component being rendered regardless of there being a title or not? Is this necessary to maintain the layout structure and is it inexpensive to rendered the Text regardless?

Food for thought, I don't mind either way, personally I don't think it's a big deal if an extra Text component is rendered in the unlikely case it has no title.

Reply:
Yes it will. that being said ,we don't want to handle that case. PropTypes should display proper warning if the required shape isn't present